underscore.js

```diff
@@ -441,7 +441,7 @@
   // Return the number of elements in an object.
   _.size = function(obj) {
     if (obj == null) return 0;
-    return isArrayLike(obj) ? obj.length : _.keys(obj).length;
+    return (property("size")(obj)) ? obj.size : isArrayLike(obj) ? obj.length : _.keys(obj).length;
```

Comment:
This is duck-typing which isn't great.
The master branch version of Underscore has `_.isMap` and `_.isSet`.